example-playbooks/restore_scylla_manager_backup/restore.yaml

```diff
@@ -0,0 +1,135 @@
+
+    - name: Recreate Scylla directories
+      file:
+        path: "/var/lib/scylla/{{ item }}"
+        state: directory
+        owner: scylla
+        group: scylla
+        mode: 0755
+      loop:
+        - data
+        - commitlog
+        - hints
+        - view_hints
+
+    - name: Get seeds list
+      shell: |
+        grep '\- seeds:' /etc/scylla/scylla.yaml | awk '{ print $NF }' | tr -d '"' | tr -d "'"
+      register: _seeds
+
+    - name: Get initial token
+      shell: |
+        scylla-manager-agent download-files -L {{ backup_location }} -n {{ host_id[inventory_hostname] }} -T {{ snapshot_tag }} --dump-tokens
+      register: _initial_token
+
+    - name: Set initial_token in scylla.yaml
+      lineinfile:
+        path: /etc/scylla/scylla.yaml
+        regexp: '^(#\s*)?initial_token:'
+        line: "initial_token: {{ _initial_token.stdout }}"
+        create: yes
+
+    - name: Make each node a seed
+      lineinfile:
+        path: /etc/scylla/scylla.yaml
+        regexp: '^(\s+) - seeds:'
+        line: |
+          \g<1> - seeds: {{ groups.all | join(',') }}
+        backrefs: yes
+
+    - name: Start Scylla service
+      service:
+        name: scylla-server
+        state: started
+
+    - name: Wait for CQL port
+      wait_for:
+        port: 9042
+
+    - name: Stop Scylla service
+      service:
+        name: scylla-server
+        state: stopped
+
+- name: Restore schema
+  hosts: all[0]
+  become: true
+  gather_facts: false
+  tasks:
+    - name: Remove other seeds
+      lineinfile:
+        path: /etc/scylla/scylla.yaml
+        regexp: '^(\s+) - seeds:'
+        line: |
+          \g<1> - seeds: {{ inventory_hostname }}
+        backrefs: yes
+
+    - name: Download system_schema
+      shell: |
+        scylla-manager-agent download-files -L {{ backup_location }} -n {{ host_id[inventory_hostname] }} -T {{ snapshot_tag }} -d /var/lib/scylla/data/ -K system_schema --silent
+      become_user: scylla
+
+    - name: Start Scylla service
+      service:
+        name: scylla-server
+        state: started
+
+- name: Restore data
+  hosts: all
+  become: true
+  gather_facts: false
+  tasks:
+    - name: Start Scylla service
+      service:
+        name: scylla-server
+        state: started
+
+    - name: Wait for CQL port
+      wait_for:
+        port: 9042
+
+    - name: Stop Scylla service
+      service:
+        name: scylla-server
+        state: stopped
+
+    - name: Download data
+      shell: |
+        scylla-manager-agent download-files -L {{ backup_location }} -n {{ host_id[inventory_hostname] }} -T {{ snapshot_tag }} -d /var/lib/scylla/data/ -K '*,!system_schema' --silent
+      become_user: scylla
+      async: 86400
+      poll: 5
+
+    - name: Resore seeds list
+      lineinfile:
+        path: /etc/scylla/scylla.yaml
+        regexp: '^(\s+) - seeds:'
+        line: |
+          \g<1> - seeds: {{ _seeds.stdout }}
+        backrefs: yes
+
+    - name: Start Scylla service
```

Comment:
fwiw I merged anyways
but ideally seeds should be started first
ala https://github.com/scylladb/scylla-ansible-roles/blob/master/example-playbooks/rolling_ops/rolling_restart.yml

we should make such OPS as whole cluster start available as some module command ... what do you think @sitano ?

Reply:
I saw that the playbook sets ALL nodes to seeds at some point when I skimmed through the new play.


PS/offtop - why not use the unirestore utility?

Reply:
unirestore is not built for restoring with initial_token (so cloning from backup with restoring backup 1-1 , no need to cleanup or upload everything everywhere), though @vladzcloudius has a patch to get this support into unirestore too

also this depends heavily on next release of manager that supports dumping initial token

Reply:
@tarzanek Let's not rush merging things before all reviewers acked the PR. 
@dyasny It's not that unirestore couldn't be used here - it could. 
This playbook is a contribution from Michal - which is definitely welcome.
One nice thing it does is that it recreates the destination cluster which is beyond the unirestore scope. 

The ultimate plan is to re-write unirestore.py logic in Go and contribute it to SM.

In general it's not the best idea to mix deployment and copying things in the same PB - better have these as separate modules and then have a PB that calls these modules.